src/main/java/util/TickingTimerManager.java

```diff
@@ -0,0 +1,100 @@
+package util;
+
+/**
+ * This class manages the TickingTimer instance used for refreshing. It computes the refresh rate period and change
+ * the refresh period.
+ */
+public class TickingTimerManager {
+
+    /**
+     * To allow some additional time until next refresh.
+     */
+    public static final int BUFFER_TIME = 1;
+
+    /**
+     * The amount that is set aside for manual refresh or creation of issues, labels and milestones.
+     */
+    public static final int APIQUOTA_BUFFER = 200;
+
+    public static final int NO_OF_REFRESH_THAT_REQUIRES_MORE_TIME = 1;
```

Comment:
Are you talking about DEFAULT_REFRESH_PERIOD_IN_SEC? Even initialising the period to default value. At some times, we still need to set back to the default.
```

  boolean isDuringAppInit = apiQuota != 0 && lastApiCallsUsed == 0;
        if (isDuringAppInit || remainingTime == 0) {
            refreshTimeInMin = Utility.secsToMins(DEFAULT_REFRESH_PERIOD_IN_SEC);
            return refreshTimeInMin;
        }

```